### ISSUE TITLE
[Bug] CEPGP tainting Raid Frames

### ISSUE THREAD

**Issue by Anilusion** [Alumian/CEPGP-Retail#245]
**The Error**
![WowClassic_2020-04-15_17-57-12](https://user-images.githubusercontent.com/9070713/79447324-fd295380-7fdf-11ea-9ec1-2b7f5ff6be8f.png)

**Describe the bug**
As of one of the more recent updates (1.12.9 did not have this bug), CEPGP is tainting the raid frames, causing them to break if changes to the group are made while in combat. This is extremely bad for an addon which purpose is to be used in a raid environment. Taint issues can be very tricky to track down, but this must be a priority.

**To Reproduce**
Have people join or leave your raid while you are in combat, and the raid frames will become desynced. Some frames will target the wrong player and other frames will not target anything at all.

**Comment by Alumian** [comment]
Can you please complete the following steps:
1. Type `/console taintLog 1`
2. Reload your UI,
3. Wait for the issue to occur again
4. Close your game
5. Navigate to \_classic\_\Logs\ and upload the file taint.log to this ticket

Thanks

**Comment by Anilusion** [comment]
[taint.log](https://github.com/Alumian/CEPGP-Retail/files/4489966/taint.log)

**Comment by Alumian** [comment]
Thanks Anilusion.
Unfortunately, that log isn't really telling me much. Can you please try again but bump up the logging to level to `/console taintLog 2`?

**Comment by Anilusion** [comment]
[taint.log](https://github.com/Alumian/CEPGP-Retail/files/4490137/taint.log)

**Comment by Alumian** [comment]
Thanks! Investigating.

**Comment by Alumian** [comment]
Thanks @Anilusion for your bug report. The taint has been rectified and I will push an update soon.

**Comment by Anilusion** [comment]
Thanks for the very fast response! Do you mind explaning what the issue was and how you tracked it down?

**Comment by Alumian** [comment]
Sure - Based on the destination of the taint, I found that it was impacting the raid frames. Although CEPGP actually has no interaction with the raid frames, I figured it had to be something related to raids.
When you said that it only happens when a player joins/leaves, my attention turned to roster updates since that is prompted when a player joins/leaves a raid. After spending a while fiddling around with that, I found that I was way off lol. But, while I was playing around with this, I just ended up commenting out a whole heap of code to try and identify the area it could be in. I found that the issue was in the options interface, I was initialising two dropdown menus. The only difference between this version and previous versions was that there was a delay. I moved it out of the delay and that fixed the taint. To my understanding, because of the delay, it caused the initialisation function to redirect to a blizzard dropdown frame which loaded after CEPGP and thus, tainted the dropdown and all subsequent elements leading down to the raid frames.
